.github/workflows/pull-request.yaml

```diff
@@ -1,13 +1,63 @@
-name: CI-pullrequest
+name: Pull request checks
 
 on: [pull_request]
 
 jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.22'
+
+      - name: Run unit tests
+        run: go test ./...
+
+  # Linting currently fails. If in the future we fix linter issues,
+  # we can enable this step.
+  # lint:
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - uses: actions/checkout@v4
+
+  #     - uses: actions/setup-go@v4
+  #       with:
+  #         go-version: '1.22'
+
+  #     - name: Run linter
+  #       uses: golangci/golangci-lint-action@v5
+  #       with:
+  #         version: v1.57
+
   build:
     runs-on: ubuntu-latest
-    container: 
-      image: rancher/dapper:v0.5.4
     steps:
-      - uses: actions/checkout@v1
-      - name: Run CI
-        run: dapper ci
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.22'
```

Comment:
The code is currently using go1.19, I think we should use the same version here to avoid any possible problems